mttl/models/containers/expert_containers.py

```diff
@@ -21,23 +23,32 @@
 from mttl.models.modifiers.modify_model import get_modifier_type
 
 
-class ExpertContainer:
+class Container(abc.ABC):
+    @abc.abstractmethod
+    def __getitem__(self, key):
+        pass
+
+    @abc.abstractmethod
+    def __len__(self):
+        pass
+
+
+class ExpertContainer(Container):
     __supports_configs__ = []
 
-    def __init__(self, config, info_container, layer, selector=None):
+    def __init__(self, config, layer, selector=None):
         from mttl.models.containers.selectors import TaskNameSelector
 
         self.config = config
         self.layer = layer
-        self.info_container = info_container
-        self.selector = selector or TaskNameSelector(info_container)
+        self.selector = selector or TaskNameSelector()
 
-        self.experts: dict = None
         self.expert_infos = {}
         self.expert_names = []
```

Comment:
Since we are going for dicts now, what about `expert_names` being a property that returns `self.expert_infos.keys()`? That would avoid any need to keep track of two different things.

Reply:
expert_names keep track of the order in which the experts are inserted, are dicts order sensitive in python 3.X?

Reply:
nice, ok cool

Reply:
Yes, dicts are ordered (insertion) since Python 3.6+. mttl requires python 3.9+.

Reply:
done